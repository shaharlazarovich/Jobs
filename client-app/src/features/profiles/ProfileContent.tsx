import React from 'react'
import {Tab} from 'semantic-ui-react'
import ProfileDescription from './ProfileDescription'

interface IProps {
    setActiveTab: (activeIndex: any) => void;
}

const panes = [
    {menuItem: 'About', render: () => <ProfileDescription />}
]
const ProfileContent: React.FC<IProps> = ({setActiveTab}) => {
    return (
        <Tab 
            data-test="component-profile-content"
            menu={{fluid: true, vertical: true}}
            menuPosition='right'
            panes={panes}
            onTabChange={(e, data) => setActiveTab(data.activeIndex)}
        />
    )
}

export default ProfileContent;
