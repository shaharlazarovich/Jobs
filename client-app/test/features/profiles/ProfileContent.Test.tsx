import { shallow } from "enzyme";
import React from "react";
import { findByTestAttr } from '../../common/testUtils';
import ProfileContent from "../../../src/features/profiles/ProfileContent";

interface IProps {
    setActiveTab: (activeIndex: any) => void;
}

const setup = (setActiveTab:any) => {
    const wrapper = shallow<IProps>(<ProfileContent {...setActiveTab} />)
    return wrapper;
}

describe('Profile Content UniTest', () => {
    let wrapper:any;
    beforeEach(() => {
        wrapper = setup([]);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-profile-content');
        expect(component.length).toBe(1);
    });
})