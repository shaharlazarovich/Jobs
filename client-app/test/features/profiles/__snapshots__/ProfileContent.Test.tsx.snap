// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Content UniTest should match the snapshot 1`] = `
<Tab
  data-test="component-profile-content"
  grid={
    Object {
      "paneWidth": 12,
      "tabWidth": 4,
    }
  }
  menu={
    Object {
      "fluid": true,
      "vertical": true,
    }
  }
  menuPosition="right"
  onTabChange={[Function]}
  panes={
    Array [
      Object {
        "menuItem": "About",
        "render": [Function],
      },
    ]
  }
  renderActiveOnly={true}
>
  <div
    data-test="component-profile-content"
  >
    <Grid>
      <div
        className="ui grid"
      >
        <GridColumn
          stretched={true}
          width={12}
        >
          <div
            className="stretched twelve wide column"
          >
            <ProfileDescription>
              <TabPane
                active={true}
                as={[Function]}
                data-test="component-profile-description"
              >
                <Segment
                  attached="bottom"
                  className="active tab"
                  data-test="component-profile-description"
                >
                  <div
                    className="ui bottom attached segment active tab"
                    data-test="component-profile-description"
                  >
                    <Grid>
                      <div
                        className="ui grid"
                      >
                        <GridColumn
                          width={16}
                        >
                          <div
                            className="sixteen wide column"
                          >
                            <Header
                              content="About Fake Username"
                              floated="left"
                              icon="user"
                            >
                              <div
                                className="ui left floated header"
                              >
                                <Icon
                                  as="i"
                                  name="user"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="user icon"
                                    onClick={[Function]}
                                  />
                                </Icon>
                                <HeaderContent>
                                  <div
                                    className="content"
                                  >
                                    About Fake Username
                                  </div>
                                </HeaderContent>
                              </div>
                            </Header>
                          </div>
                        </GridColumn>
                        <GridColumn
                          width={16}
                        >
                          <div
                            className="sixteen wide column"
                          >
                            <span />
                          </div>
                        </GridColumn>
                      </div>
                    </Grid>
                  </div>
                </Segment>
              </TabPane>
            </ProfileDescription>
          </div>
        </GridColumn>
        <GridColumn
          width={4}
        >
          <div
            className="four wide column"
          >
            <Menu
              activeIndex={0}
              fluid={true}
              items={
                Array [
                  "About",
                ]
              }
              onItemClick={[Function]}
              vertical={true}
            >
              <div
                className="ui fluid vertical menu"
              >
                <MenuItem
                  active={true}
                  content="About"
                  index={0}
                  key="About"
                  name="About"
                  onClick={[Function]}
                >
                  <a
                    className="active item"
                    onClick={[Function]}
                  >
                    About
                  </a>
                </MenuItem>
              </div>
            </Menu>
          </div>
        </GridColumn>
      </div>
    </Grid>
  </div>
</Tab>
`;
